node/test_force.js

```diff
@@ -0,0 +1,383 @@
+            // Pointing to repoDir in Podfile
+            shelljs.sed('-i', /pod ('Salesforce.*')/g, 'pod $1, :path => \'../SalesforceMobileSDK-iOS\'', path.join(appDir, 'Podfile'));
+            shelljs.sed('-i', /pod ('Smart.*')/g, 'pod $1, :path => \'../SalesforceMobileSDK-iOS\'', path.join(appDir, 'Podfile'));
+
+            shelljs.pushd(appDir);
+            runProcessCatchError('pod update');    
+            shelljs.popd();
+
+            var workspacePath = path.join(appDir, appName + '.xcworkspace');
+            runProcessCatchError('xcodebuild -workspace ' + workspacePath + ' -scheme Pods-' + appName + ' clean build', 'COMPILING ' + target)
+        }
+        else {
+            shelljs.pushd(appDir);
+            runProcessCatchError('cordova build', 'COMPILING ' + target);    
+            shelljs.popd();
+        }
+        
+    }
+    else if (os === OS.android) {
+        var targetDir;
+
+        if (appType.indexOf('native')>=0) {
+            shelljs.mkdir('-p', appDir);
+            targetDir = appDir;
+        }
+        else {
+            targetDir = tmpDir;
+        }
+
+        var forcedroidArgs = 'create '
+            + ' --apptype=' + appType
+            + ' --appname=' + appName
+            + ' --packagename=com.mycompany'
+            + ' --targetdir=' + targetDir
+            + ' --usesmartstore=yes';
+            if (appType === 'hybrid_remote') {
+                forcedroidArgs += ' --startpage=/apex/testPage';
+            }
+
+        runProcessCatchError(forcePath + ' ' + forcedroidArgs, 'GENERATING ' + target);
+
+        if (appType.indexOf('native')>=0) {
+            shelljs.pushd(appDir);
+            runProcessCatchError('./gradlew', 'COMPILING ' + target);    
+            shelljs.popd();
+        }
+        else {
+            shelljs.pushd(appDir);
+            runProcessCatchError('cordova build', 'COMPILING ' + target);    
+            shelljs.popd();
+        }
+    }
+}
+
+//
+// Update cordova plugin repo
+//
+function updatePluginRepo(tmpDir, pluginRepoDir, branch) {
+    log('Updating cordova plugin at ' + branch, COLOR.green);
+    shelljs.pushd(pluginRepoDir);
+    runProcessThrowError(path.join('tools', 'update.sh') + ' -b ' + branch);    
+    shelljs.popd();
+}
+
+//
+// Update forceios/forcedroid script to use local plugin repo
+//
+function editForceScriptToUseLocalPluginRepo(tmpDir, os) {
+    log('Editing  ' + forcePackageNameForOs(os) + '.js to use local cordova plugin', COLOR.green);
+    shelljs.sed('-i', /'cordova plugin add .*'/g, '\'cordova plugin add ../SalesforceMobileSDK-CordovaPlugin\'', path.join(tmpDir, 'node_modules', forcePackageNameForOs(os), 'node', forcePackageNameForOs(os) + '.js'));
+}    
+
+//
+// Helper to run arbitrary shell command - errors caught (and reported)
+//
+function runProcessCatchError(cmd, msg) {
+    try {
+        if (msg) log('!TESTING!  ' + msg, COLOR.yellow);
+        log('Running: ' + cmd);
+        execSync(cmd);
```

Comment:
Are we sure that `execSync` will throw an exception whenever a process exits with a non-zero exit code?  Running this script against iOS, I got errors back:
```
Running: xcodebuild -workspace tmp/testforce855/nativeiosApp/nativeiosApp.xcworkspace -scheme Pods-nativeiosApp clean build
2016-02-04 09:42:07.615 xcodebuild[21345:414749]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTFrameworks/DVTFrameworks-9073/DVTFoundation/Utilities/DVTFileSystemUtilities.mm:142
Details:  lstat('/Users/khawkins/Library/Developer/Xcode/DerivedData/nativeiosApp-cpzpeorkqsgtgacffubeatqpqzyf/Build/Products/Debug-iphoneos/SalesforceSDKResources.bundle') failed with errno 2 (No such file or directory)
Function: DVTRecursiveMkdirResult dvt_recursiveMkdir_returningErrorString(NSString *__strong, NSString *__autoreleasing *)
Thread:   <NSThread: 0x7fb76d443dd0>{number = 9, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
```
But the script reported as successful.